### PR TITLE
API

### PR DESCRIPTION
**Pull request by SamuelTakyiBoateng**
## BookStore (Connect to API)
-  Bookstore API was used as the backend of the application.
- The  axios was used with createAsyncThunk to make the HTTP requests to the backend
- The books can be fetched from the Bookstore API after the component has rendered
- The add book and delete book features was refactored to persist obtained changes in the server.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,22 +1,77 @@
...
+  });
+  return result;
+};
+
+export const getBookItems = createAsyncThunk('books/getBookItems', async () => {
+  try {
+    const response = await axios(`${baseUrl}/apps/${appId}/books`);
+    return response.data;
+  } catch (err) {
+    console.error('Error fetching data');
```
Comment (Alusp) [inline src/redux/books/booksSlice.js:27]: - [ ] Good job far, :clap: however, please ensure that you remove console.log from your code base, Including console.log statements in your code can negatively impact code reviews and overall code quality. Code with excessive or unnecessary logging can be seen as less professional and harder to maintain. :slightly_smiling_face:

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,22 +1,77 @@
...
+    throw err;
+  }
+});
+
+const postBookItem = async (newBook) => {
+  try {
+    await axios.post(`${baseUrl}/apps/${appId}/books`, newBook);
+  } catch (error) {
+    console.error('Error adding book:', error);
```
Comment (Alusp) [inline src/redux/books/booksSlice.js:36]: - [ ] Good job far, :clap: however, please ensure that you remove console.log from your code base, Including console.log statements in your code can negatively impact code reviews and overall code quality. Code with excessive or unnecessary logging can be seen as less professional and harder to maintain. :slightly_smiling_face:

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -1,22 +1,77 @@
...
+    throw error;
+  }
+};
+
+const deleteBookItem = async (bookId) => {
+  try {
+    await axios.delete(`${baseUrl}/apps/${appId}/books/${bookId}`);
+  } catch (error) {
+    console.error('Error deleting book:', error);
```
Comment (Alusp) [inline src/redux/books/booksSlice.js:45]: - [ ] Good job far, :clap: however, please ensure that you remove console.log from your code base, Including console.log statements in your code can negatively impact code reviews and overall code quality. Code with excessive or unnecessary logging can be seen as less professional and harder to maintain. :slightly_smiling_face:

**Review by Alusp — CHANGES_REQUESTED** [review]
Hi @SamuelTakyiBoateng, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
close

 ## Highlights
✔️ Understand the concept of state management. ✔️
✔️ Use store, actions and reducers in React.. ✔️
✔️ Tasks almost complete  ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @ALusp in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 3 above)
Comment (SamuelTakyiBoateng) [reply src/redux/books/booksSlice.js]: Thank you so much for meticulous review.
Please, the proposed changes have been effected. Thanks again!

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (SamuelTakyiBoateng) [reply src/redux/books/booksSlice.js]: Thank you so much for meticulous review.
Please, the proposed changes have been effected. Thanks again!

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (SamuelTakyiBoateng) [reply src/redux/books/booksSlice.js]: Thank you so much for meticulous review.
Please, the proposed changes have been effected. Thanks again!

**Review by Reem-lab — APPROVED** [review]
# Hi @SamuelTakyiBoateng,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on the API🤩
- [ ]  No Linters error ✅


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**